src/Raven.Client/Documents/Operations/Replication/FeatureTaskDefinition.cs

```diff
@@ -33,21 +38,35 @@ public virtual DynamicJsonValue ToJson()
             };
         }
 
-        public void Validate(string thumbprint)
+        public bool CanAccess(string thumbprint, out string err)
         {
+            err = null;
             if (string.IsNullOrEmpty(thumbprint))
             {
                 if (Certificates == null || Certificates.Count == 0)
-                    return;
+                    return true;
 
-                throw new AuthorizationException($"Certificate is needed for pull replication '{Name}'");
+                err = $"Certificate is needed for pull replication '{Name}'";
+                return false;
             }
 
-            if (Certificates.Contains(thumbprint) == false)
+            foreach (var certificate in Certificates)
             {
-                throw new AuthorizationException($"Certificate with the thumbprint {thumbprint} was not found for pull replication '{Name}'");
+                var cert = new X509Certificate2(Convert.FromBase64String(certificate));
```

Comment:
Just validating the thumbprint is fine, need to note here that we rely on calling code to validate the actual cert (expiration, etc)